### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by martymichal**
This is a combination of GitHub's .gitignore's for Python[0] and JetBrains[1] and a bit of Perun on top..

[0] https://github.com/github/gitignore/blob/main/Python.gitignore
[1] https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,240 @@
...
+#  JetBrains specific template is maintained in a separate JetBrains.gitignore that can
+#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+#  and can be added to the global gitignore or merged into this file.  For a more nuclear
+#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
+#.idea/
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
```
Comment (tfiedor) [inline .gitignore]: Please, lets merge this to `.idea/**` or `.idea/*`. I don't think there is valid reason for not ignoring anything related to IDEs.

**Review by tfiedor — CHANGES_REQUESTED** [review]
I think the new `.gitignore` is good (I guess generated by some tool), but I would squash the idea directories (and maybe, if there are more similar cases that I overlooked, I would squash them as well; I'd rather keep the stuff short and concise).

File: .gitignore (hunk 1 above)
Comment (JiriPavela) [reply .gitignore]: I agree. There is an explanation (in the long comment just above the shown snippet) of why the .idea ignores are scattered like that, but I think ignoring the entire .idea shouldn't really introduce any problem.

Comment (martymichal) [reply .gitignore]: Done.

**Review by tfiedor — APPROVED** [review]

**Review by JiriPavela — APPROVED** [review]

**Pull request merged**